detectors/node/opentelemetry-resource-detector-aws/src/detectors/AwsEcsDetector.ts

```diff
@@ -77,13 +95,149 @@ export class AwsEcsDetector implements Detector {
       const splitData = rawData.trim().split('\n');
       for (const str of splitData) {
         if (str.length > this.CONTAINER_ID_LENGTH) {
-          return str.substring(str.length - this.CONTAINER_ID_LENGTH);
+          containerId = str.substring(str.length - this.CONTAINER_ID_LENGTH);
+          break;
         }
       }
     } catch (e) {
```

Comment:
Add type?

Reply:
It actually seems that, rather than interpolating the message, the best practice is to pass the error as the second argument of the logger call. Done in ea3ea2ca.